model/PerpendicularPaths.py

```diff
@@ -71,33 +71,29 @@ def robot_by_cell (self, cell):
                 return r;
 
     def space_touched_by_xy (self, cell):
-        color = 15
+        color = 15 if os.name == 'nt' else ''
         for i, s in enumerate(self.space_touched):
             if s[1] == cell:
                 color = s[2]
         return color
 
+    def color_update (self, color, bgcolor=15 if os.name == 'nt' else 0):
+        if os.name == 'nt':
+            if self.std_output_hdl is None:
+                STD_OUTPUT_HANDLE_ID = c_ulong(0xfffffff5)
+                windll.Kernel32.GetStdHandle.restype = c_ulong
+                std_output_hdl = windll.Kernel32.GetStdHandle(STD_OUTPUT_HANDLE_ID)
+            windll.Kernel32.SetConsoleTextAttribute(self.std_output_hdl, bgcolor | color)
+        else:
+            print ("\033[" + str(bgcolor) + str(color) + "m", end="")
+
+
     def display_update(self):
         #print (self.rr_board)
-        #WINDOWS SPECIFIC TERMINAL COLORS
-        if os.name == 'nt':
-            STD_OUTPUT_HANDLE_ID = c_ulong(0xfffffff5)
-            windll.Kernel32.GetStdHandle.restype = c_ulong
-            std_output_hdl = windll.Kernel32.GetStdHandle(STD_OUTPUT_HANDLE_ID)
-            windll.Kernel32.SetConsoleTextAttribute(std_output_hdl, 15 | 0x0050)
         goal = self.board_section.goals[self.goal_index]
-        print ("Goal " + str(self.goal_index+1) + " of " + str(len(self.board_section.goals)), end="", flush=True)
-        windll.Kernel32.SetConsoleTextAttribute(std_output_hdl, 15)
-        print (": move ", end="", flush=True)
-        #TODO: this is saying only highlight the text the color of the first robot
-        # need to fix for multi goal
-        windll.Kernel32.SetConsoleTextAttribute(std_output_hdl, 15 | goal.robots[0].bgcolor())
-        print (goal.robots[0].name, end="", flush=True)
-        windll.Kernel32.SetConsoleTextAttribute(std_output_hdl, 15)
-        print (" to cell [", end="", flush=True)
-        windll.Kernel32.SetConsoleTextAttribute(std_output_hdl, 15 | 0x0050)
-        print (str(goal.point.x) + "," + str(goal.point.y), end="", flush=True)
-        windll.Kernel32.SetConsoleTextAttribute(std_output_hdl, 15)
+        print ("Goal " + str(self.goal_index+1) + " of " + str(len(self.board_section.goals)) + 
+        ": move " + goal.robots[0].name + " to cell [" + (str(goal.point.x)) + "," + 
+        str(goal.point.y), end="", flush=True)
         print ("]\r\n\t", end="")
```

Comment:
new line 97 can be added to the above print statement , its just a continuation of the goal information